frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
+                            case columnTitle === ColumnDataEnum.Name:
+                                if (receiverItem.receivingOrg)
+                                    formattedData.Name =
+                                        receiverItem.receivingOrg;
+                                break;
+                            case columnTitle === ColumnDataEnum.Service:
+                                if (receiverItem.receivingOrgSvc)
+                                    formattedData.Service =
+                                        receiverItem.receivingOrgSvc;
+                                break;
+                            case columnTitle === ColumnDataEnum.Date:
+                                if (receiverItem.createdAt)
+                                    formattedData.Date =
+                                        dateTimeFormatter.format(
+                                            new Date(receiverItem.createdAt)
+                                        );
+
+                                break;
+                            case columnTitle === ColumnDataEnum.ReportId:
+                                if (receiverItem.reportId)
+                                    formattedData.ReportId =
+                                        receiverItem.reportId;
+                                break;
+                            case columnTitle === ColumnDataEnum.Main:
+                                const { folderLocation } = parseFileLocation(
+                                    receiverItem?.fileUrl || ""
+                                );
+                                formattedData.Main = folderLocation;
+                                break;
+                            case columnTitle === ColumnDataEnum.Sub:
+                                const { sendingOrg } = parseFileLocation(
+                                    receiverItem?.fileUrl || ""
+                                );
+                                formattedData.Sub = sendingOrg;
+                                break;
+                            case columnTitle === ColumnDataEnum.FileName:
+                                const { fileName } = parseFileLocation(
+                                    receiverItem?.fileUrl || ""
+                                );
+                                formattedData.FileName = fileName;
+                                break;
+                            case columnTitle ===
+                                ColumnDataEnum.TransportResults:
+                                if (receiverItem.transportResult)
+                                    formattedData.TransportResults =
+                                        receiverItem.transportResult;
+                                break;
+                        }
+                    }
+                    return formattedData;
+                }
+            ),
+        [receiverDetails]
+    );
+    const sortedData = useMemo(
```

Comment:
I think this could also be generalized with the [data format](https://github.com/CDCgov/prime-reportstream/pull/8295/files#r1108850755), no?  Like if someone clicks on the 0th header, we'd just sort the array of tuples.  The generalized sort function could include type checking and/or casting as needed 🤔

Reply:
Added to our discussion on what features to add for a genericized table component: https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8280